components/builder-protocol/protocols/scheduler.proto

```diff
@@ -0,0 +1,20 @@
+package scheduler;
+import "protocols/depotsrv.proto";
+
+message Schedule {
+  required depotsrv.PackageIdent ident = 1;
```

Comment:
I've been using `required` in my protocol definitions but I don't know if that's the right thing to do. There is no such thing as `required` in proto3 (we're still using 2) anymore and there are [warnings about it's use](https://developers.google.com/protocol-buffers/docs/proto) in some of the protobuf docs. Adam read those warnings and began using `optional` only in his protobuf files.

I think it'd be valuable for you to read up on this and help us make a decision as to if we should use `required` or just stick to `optional`. At this point I'm leaning towards using `optional` only since `required` has been removed from proto3. What do you think?